File: VB/WindowsApplication1/Classes/AutoHideBar.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars
Imports DevExpress.Utils.Drawing.Animation

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class AutoHideBar
		Inherits Bar


		Private _EventSubscriber As EventSubscriber
		Public ReadOnly Property EventSubscriber() As EventSubscriber
			Get
				If _EventSubscriber Is Nothing Then
					_EventSubscriber = New EventSubscriber(Me)
				End If
				Return _EventSubscriber
			End Get
		End Property

		Private ReadOnly Overloads Property DockControl() As BarDockControl
			Get
				Return Manager.DockControls(DockStyle)
			End Get
		End Property
		Public Sub New(ByVal manager As BarManager, ByVal style As BarDockStyle)
			MyBase.New(manager)
			If style = BarDockStyle.Bottom Then
				DockStyle = BarDockStyle.Bottom
			Else
				DockStyle = BarDockStyle.Top
			End If
			OptionsBar.AllowCollapse = True
			OptionsBar.DrawDragBorder = False
			OptionsBar.AllowQuickCustomization = False
		End Sub

		Protected Overrides Function CalcBarEndSize() As Size
			If OptionsBar.BarState = BarState.Expanded Then
				Return MyBase.CalcBarEndSize()
			End If
			Dim size As Size = BarControl.Bounds.Size
			size.Height = 5
			Return size
		End Function

		Public Sub HideBar()
			OptionsBar.BarState = BarState.Collapsed
		End Sub

		Public Sub ShowBar()
			OptionsBar.BarState = BarState.Expanded
		End Sub

		Protected Overrides Sub OnBarChanged()
			MyBase.OnBarChanged()
			If (Not DesignMode) Then
				EventSubscriber.CheckChange(BarControl, DockControl)
			End If
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/Classes/AutoHideBar.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars;
using DevExpress.Utils.Drawing.Animation;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class AutoHideBar : Bar
    {


        private EventSubscriber _EventSubscriber;
        public EventSubscriber EventSubscriber
        {
            get
            {
                if (_EventSubscriber == null)
                    _EventSubscriber = new EventSubscriber(this);
                return _EventSubscriber;
            }
        }

        private BarDockControl DockControl
        {
            get
            {
                return Manager.DockControls[DockStyle];
            }
        }
        public AutoHideBar(BarManager manager, BarDockStyle style)
            : base(manager)
        {
            if (style == BarDockStyle.Bottom)
                DockStyle = BarDockStyle.Bottom;
            else
                DockStyle = BarDockStyle.Top;
            OptionsBar.AllowCollapse = true;
            OptionsBar.DrawDragBorder = false;
            OptionsBar.AllowQuickCustomization = false;
        }

        protected override Size CalcBarEndSize()
        {
            if (OptionsBar.BarState == BarState.Expanded)
                return base.CalcBarEndSize();
            Size size = BarControl.Bounds.Size;
            size.Height = 5;
            return size;
        }

        public void HideBar()
        {
            OptionsBar.BarState = BarState.Collapsed;
        }

        public void ShowBar()
        {
            OptionsBar.BarState = BarState.Expanded;
        }

        protected override void OnBarChanged()
        {
            base.OnBarChanged();
            if (!DesignMode)
                EventSubscriber.CheckChange(BarControl, DockControl);
        }
    }
}